e2e/auth.setup.ts

```diff
@@ -0,0 +1,110 @@
+// Required environment variables:
+// - ADMIN_USERNAME: username for admin
+// - ADMIN_PASSWORD: password for admin
+//
+// Environment variables:
+// - in CI: set in repo GitHub secrets, read via env in workflow
+// - locally: set in .env file, read by dotenv
+
+import { Page, expect, test as setup } from '@playwright/test'
+import { v4 as uuidv4 } from 'uuid'
+import { ADMIN_STORAGE_STATE, USER_STORAGE_STATE } from '../playwright.config'
+import { getEndpoint } from './helpers/http'
+import { setLocalStorage } from './helpers/localStorage'
+import {
+  USER_NAME_LOCALSTORAGE_KEY,
+  createTestUser,
+  deleteTestUser,
+  getAccessTokenFromCookie,
+  getAdminUserCredentials,
+  loginTestUser,
+} from './helpers/testUser'
+import { TestUser } from './helpers/types'
+
+let testUserId: string | undefined
+let adminAccessToken: string | undefined
+let adminPage: Page
+let userPage: Page
+
+setup.describe('Setup', () => {
+  setup.beforeAll(async ({ browser }) => {
+    adminPage = await browser.newPage()
+    userPage = await browser.newPage()
+  })
+
+  setup('authenticate users', async () => {
+    setup.slow()
```

Comment:
This may be the first test to run and trigger SWC compilation, which may take 15+ seconds. Mark this [test as slow](https://playwright.dev/docs/api/class-test#test-slow-1), so test will be given triple timeout.

Reply:
Can you add this as a comment above this line?